File: SDKs/twilio.com/twilio_video_v1/1.34.0/go/pkg/models/shared/videov1roomroomparticipantroomparticipantsubscriberule.go
package shared

import (
	"time"
)

type VideoV1RoomRoomParticipantRoomParticipantSubscribeRuleRules struct {
	All       *bool   `json:"all"`
	Kind      *string `json:"kind"`
	Priority  *string `json:"priority"`
	Publisher *string `json:"publisher"`
	Track     *string `json:"track"`
	Type      *string `json:"type"`
}

type VideoV1RoomRoomParticipantRoomParticipantSubscribeRule struct {
	DateCreated    *time.Time                                                    `json:"date_created"`
	DateUpdated    *time.Time                                                    `json:"date_updated"`
	ParticipantSid *string                                                       `json:"participant_sid"`
	RoomSid        *string                                                       `json:"room_sid"`
	Rules          []VideoV1RoomRoomParticipantRoomParticipantSubscribeRuleRules `json:"rules"`
}

File: SDKs/twilio.com/twilio_video_v1/1.36.0/typescript/src/sdk/models/shared/videov1roomroomparticipantroomparticipantsubscriberule.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";


export class VideoV1RoomRoomParticipantRoomParticipantSubscribeRuleRules extends SpeakeasyBase {
  @Metadata({ data: "json, name=all" })
  all?: boolean;

  @Metadata({ data: "json, name=kind" })
  kind?: string;

  @Metadata({ data: "json, name=priority" })
  priority?: string;

  @Metadata({ data: "json, name=publisher" })
  publisher?: string;

  @Metadata({ data: "json, name=track" })
  track?: string;

  @Metadata({ data: "json, name=type" })
  type?: string;
}


export class VideoV1RoomRoomParticipantRoomParticipantSubscribeRule extends SpeakeasyBase {
  @Metadata({ data: "json, name=date_created" })
  dateCreated?: Date;

  @Metadata({ data: "json, name=date_updated" })
  dateUpdated?: Date;

  @Metadata({ data: "json, name=participant_sid" })
  participantSid?: string;

  @Metadata({ data: "json, name=room_sid" })
  roomSid?: string;

  @Metadata({ data: "json, name=rules", elemType: shared.VideoV1RoomRoomParticipantRoomParticipantSubscribeRuleRules })
  rules?: VideoV1RoomRoomParticipantRoomParticipantSubscribeRuleRules[];
}
